stdlib/public/SwiftShims/RefCount.h

```diff
@@ -273,6 +349,44 @@ class StrongRefCount {
     return __atomic_compare_exchange(&refCount, &oldval, &newval, 0,
                                      __ATOMIC_ACQUIRE, __ATOMIC_RELAXED);
   }
+
+  template <bool ClearPinnedFlag>
+  bool doDecrementShouldDeallocateNNonAtomic(uint32_t n) {
+    // If we're being asked to clear the pinned flag, we can assume
+    // it's already set.
+    uint32_t delta = (n << RC_FLAGS_COUNT) + (ClearPinnedFlag ? RC_PINNED_FLAG : 0);
+    uint32_t val = __atomic_load_n(&refCount, __ATOMIC_RELAXED);
+    val -= delta;
+    __atomic_store_n(&refCount, val, __ATOMIC_RELAXED);
```

Comment:
I don't think this intermediate store and later assertion really helps us here, since without a barrier we aren't guaranteed to interact with updates from other threads anyway.